.github/workflows/publish.yaml

```diff
@@ -0,0 +1,22 @@
+name: publish
+
+on:
+  push:
+    branches: main
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    strategy:
```

Comment:
This is slightly less efficient than running additional steps but uses less configuration and makes the publishing steps more higher level. I can go either way.